pkg/component/gardenerapiserver/deployment.go

```diff
@@ -14,9 +14,139 @@
 
 package gardenerapiserver
 
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/utils/pointer"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/component/apiserver"
+	etcdconstants "github.com/gardener/gardener/pkg/component/etcd/constants"
+	kubeapiserverconstants "github.com/gardener/gardener/pkg/component/kubeapiserver/constants"
+	"github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector/references"
+	"github.com/gardener/gardener/pkg/utils"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+)
+
 const (
 	secretNameServer = "gardener-apiserver"
 	containerName    = "gardener-apiserver"
 
 	port = 8443
 )
+
+func (g *gardenerAPIServer) deployment(
+	secretCAETCD *corev1.Secret,
+	secretETCDClient *corev1.Secret,
+	secretGenericTokenKubeconfig *corev1.Secret,
+	secretServer *corev1.Secret,
+	secretAdmissionKubeconfigs *corev1.Secret,
+	secretETCDEncryptionConfiguration *corev1.Secret,
+	secretAuditWebhookKubeconfig *corev1.Secret,
+	secretVirtualGardenAccess *gardenerutils.AccessSecret,
+	configMapAuditPolicy *corev1.ConfigMap,
+	configMapAdmissionConfigs *corev1.ConfigMap,
+) *appsv1.Deployment {
+	deployment := &appsv1.Deployment{
```

Comment:
Will [This](https://github.com/gardener/gardener/blob/06f3b19981f6ee4fdfb2471b87a1f2cb5bb9f84e/charts/gardener/controlplane/charts/runtime/templates/apiserver/deployment.yaml#L488-L513) service still be deployed through chart?

Reply:
Nope, it is only used for local development, but productive setups don't deploy the etcd sidecar container into the GAPI pod.